server-spi-private/src/main/java/org/keycloak/crypto/SignatureProvider.java

```diff
@@ -25,6 +27,9 @@
 
     SignatureVerifierContext verifier(String kid) throws VerificationException;
 
+    // KEYCLOAK-8460 client signed signature verification
+    SignatureVerifierContext verifier(ClientModel client, JWSInput input) throws VerificationException;
```

Comment:
I see. I will prepare ClientSignatureVerifierProvider mechanism the same as what you've done for Token Signature SPI to avoid hard-cording algorithms.

Af for // KEYCLOAK-8460... , comment, I wll get rid of them.

Reply:
One point I would like to ask for some suggestion.

I've implemented ClientSignatureVerifierProvider, but I'm not sure whether the client signed JWT (ex. Request Object, currently treated as JsonNode) should implement Token interface and add corresponding entry onto TokenCategory enum.

If this ClientSignatureVerifierProvider might be used for the client signed JWT other than Request Object (ex. JWS Client Assertion for client authentication), it seems to be better to do that.

Reply:
I think that would be an improvement